meson.build

```diff
@@ -20,8 +20,10 @@ endif
 
 flatpak_intf_dir = get_option('flatpak-interfaces-dir')
 if flatpak_intf_dir == ''
-    flatpak_dep = dependency('flatpak', version: '>= 1.5.0')
-    flatpak_intf_dir = flatpak_dep.get_variable(pkgconfig: 'interfaces_dir')
+    flatpak_dep = dependency('flatpak', version: '>= 1.5.0', required : false)
```

Comment:
I think it's better to require Flatpak if `host_machine.system()` is `linux`, something like:

```meson
flatpak_required = host_machine.system() in ['linux']
flatpak_dep = dependency('flatpak', version: '>= 1.5.0', required : flatpak_required)
```

Reply:
Why? As a FreeBSD developer I hate any sorts of `if (linux)` conditions, as they usually don't check for they are really intended for. Why should we use it here? Flatpak can be never porter to non-Linux systems?

Reply:
> Flatpak can be never porter to non-Linux systems?

Correct. Flatpak relies on user namespaces, which are a Linux kernel feature and are not portable. It also makes various other Linux assumptions.

Reply:
The Flatpak dependency is required for the build system to fetch the Flatpak interfaces directory, which is necessary for building the documentation pages for xdg-desktop-portal. I asked you to make this change because it currently is required on Linux already, and I don't want to change that behavior, and I was okay with opening an exception to FreeBSD given the amount of work you've put into it. But on a second look, I think a better approach would be to error out if the build system has `docbook-docs` enabled and `flatpak-interfaces-dir` is not passed and the `flatpak` dependency is not present.

> Flatpak can be never porter to non-Linux systems?

I'm afraid that's not a question I can answer, I assumed FreeBSD developers like yourself would know how feasible such a port it.

Reply:
> > Flatpak can be never porter to non-Linux systems?
> 
> Correct. Flatpak relies on user namespaces, which are a Linux kernel feature and are not portable. It also makes various other Linux assumptions.

FreeBSD has its own primitives for containers, so there is nothing fundamentally preventing Flatpak from being ported.

> But on a second look, I think a better approach would be to error out if the build system has docbook-docs enabled and flatpak-interfaces-dir is not passed and the flatpak dependency is not present.

This would disable building any docs, not only Flatpak-related.

Reply:
> FreeBSD has its own primitives for containers, so there is nothing fundamentally preventing Flatpak from being ported

I'm sure it would be entirely possible to do something similar to Flatpak on FreeBSD, if people want to, but one of the design decisions in bubblewrap and Flatpak is that they're Linux-specific, don't aim to be portable, and don't attempt to restrict themselves to doing only portable things. x-d-p already has a small abstraction layer for multiple app frameworks, so if FreeBSD at some point has a Flatpak-like "app container" thing with the properties that x-d-p needs, then x-d-p could support that alongside Flatpak and Snap.